golang/.gitignore

```diff
@@ -3,6 +3,4 @@
 .DS_Store
 build
 Godeps/_workspace
-messagetype_string.go
-reqresreaderstate_string.go
-reqreswriterstate_string.go
+*_string.go
```

Comment:
I'll let @mmihic comment on this. I'm fine with either approach. If we decide to check in generated files, I'll do it as a separate PR.